dojo/settings/settings.dist.py

```diff
@@ -120,7 +120,7 @@
     DD_SAML2_ATTRIBUTES_MAP=(dict, {
         # Change Email/UserName/FirstName/LastName to corresponding SAML2 userprofile attributes.
         'Email': ('email', ),
-        # 'UserName': ('username', ),
+        'UserName': ('username', ),
```

Comment:
@Maffooch Does this work in your setup? In the past I had problems when mapping username here and also setting SAML_DJANGO_USER_MAIN_ATTRIBUTE = 'username'.

Reply:
My org does not have a username attribute to provide, so I put the email attribute in both the email and username field on the user
```
SAML_ATTRIBUTE_MAPPING =  {
        # Change Email/UserName/FirstName/LastName to corresponding SAML2 userprofile attributes.
        'email': ('email', 'username'),
        'firstName': ('first_name', ),
        'lastName': ('last_name', ),
    }
```

Reply:
well, users have to adopt there configuration anyway